connect/api.go

```diff
@@ -52,8 +52,21 @@ func GetProduct(productQuery Product) (Product, error) {
 		return remoteProduct, err
 	}
 	err = json.Unmarshal(resp, &remoteProduct)
+	return remoteProduct, err
+}
+
+func upgradeProduct(product Product) (Service, error) {
```

Comment:
ok, so in client.go we can create a UpgradeProduct() which simply just calls to here. That's a convention we could follow and it is closer to the ruby. What do you think?

Reply:
do we need it now? or shall we add it when it needs to be used from outside the package?

Reply:
I think deregister calls it. So we will need it for that
https://github.com/SUSE/connect/blob/master/lib/suse/connect/client.rb#L65